docs/config.md

```diff
@@ -318,6 +319,100 @@ location /blog/ {
 
 Default: `[]`
 
+- `NoDNSLink`  
+  A boolean to configure whether DNSLink lookup for value in `Host`
+  HTTP header should be performed.  If DNSLink is present, content path stored
+  in the DNS TXT record is mounted at `/` and respective payload is returned to
+  the client.
+
+Default: `false`
+
+### Gateway.PublicGateways
+
+`PublicGateways` is a dictionary for defining gateway behavior on specific fully qualified domain names (FQDN).  
+Options for changing behavior per hostname:
+
+- `Paths`  
+  Array of paths that should be mounted at the root of domain name.  
+  Example: `["/ipfs", "/ipns", "/api"]`  
+  - **Note:** when both `Paths` and DNSLink are mounted, `Paths` takes priority
+
+- `UseSubdomains`
+  A boolean to configure whether the gateway provides [Origin isolation](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)
+  between content roots.  
+  Default: `false`
+  - `true` – mount [subdomain gateway](#https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `http://{cid}.ipfs.{hostname}`
+    - requires respective `Paths` to be set
+    - requests for `http://{hostname}/ipfs/{cid}` will return redirect to `http://{cid}.ipfs.{hostname}`
+  - `false` – mount [path gateway](https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#path-gateway) at `http://{hostname}/ipfs/{cid}`
+    <!-- **(not implemented yet)** due to the lack of Origin isolation, cookies and storage on `Paths` will be disabled by [Clear-Site-Data](https://github.com/ipfs/in-web-browsers/issues/157) header -->
+
+#### Examples of common use cases
+
+* Public [subdomain gateway](https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `http://{cid}.ipfs.dweb.link`
+   ```json
+   "Gateway": {
+       "PublicGateways": {
+       "dweb.link": {
+           "UseSubdomains": true,
+           "Paths": ["/ipfs", "/ipns"]
+   ```
+   **Note:** this enables automatic redirects from content paths to subdomains  
+   `http://dweb.link/ipfs/{cid}` → `http://{cid}.ipfs.dweb.link`
+
+* Public [path gateway](https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#path-gateway) at `http://ipfs.io/ipfs/{cid}`
+   ```json
+   "Gateway": {
+     "PublicGateways": {
+       "ipfs.io": {
+         "UseSubdomains": false,
+         "Paths": ["/ipfs", "/ipns", "/api"]
+   ```
+
+* Public [DNSLink](https://dnslink.io/) gateway resolving every hostname passed in `Host` header
+  (this is the default, works out of the box):
+   ```json
+   "Gateway": {
+     "NoDNSLink": false
+   ```
+
+* Hardened, site-specific [DNSLink](https://dnslink.io/) gateway.  
+  Disabling public gateway by default (`NoFetch: true`),
+  and enabling selected gateway features only on specific hostname for which data
+  is already present on the node:
+  ```json
+  "Gateway": {
+      "NoFetch": true,
+      "NoDNSLink": true,
+      "PublicGateways": {
+        "en.wikipedia-on-ipfs.org": {
+          "NoDNSLink": false,
```

Comment:
Does this default to true or false?

Reply:
It defaults to false, so DNSLink works out of the box. 

We could simplify example and remove explicit definition, but then it would have an empty object, so I've left it to make example more clear.

Reply:
It's just a bit funny because the only place we specify this is where it's _false_. Given that, I'd assume it defaults to true.